slicedimage/backends/_http.py

```diff
@@ -10,9 +11,21 @@ class HttpBackend(Backend):
     def __init__(self, baseurl):
         self._baseurl = baseurl
 
-    def read_file_handle_callable(self, name, checksum_sha1=None):
+    def read_file_handle_callable(self, name, checksum_sha1=None, seekable=False):
         def returned_callable():
             parsed = pathjoin(self._baseurl, name)
+            if seekable:
+                req = requests.get(parsed)
+                return _BytesIOContextManager(req.content)
             req = requests.get(parsed, stream=True)
             return req.raw
         return returned_callable
+
+
+class _BytesIOContextManager(BytesIO):
```

Comment:
Unopinionated question: This seems like it might be a more general IO tool -- is this the right module for it? 

If yes, since `_BytesIOContextManager` is called by `HttpBackend`, I'd order it above `HttpBackend` in the file.

Reply:
I moved this into _base.py.  I think @shanaxel42 might need this for caching.